components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/hardware-wallet-connect/index.tsx

```diff
@@ -311,115 +311,78 @@ export const HardwareWalletConnect = ({
   }, [savedAccounts])
 
   // render
-  if (showAuthorizeDevice) {
+  if (showAccountsList && selectedHardwareWallet) {
     return (
       <>
-        <HardwareInfoRow>
-          <InfoIcon />
-          <HardwareInfoColumn>
-            <DisclaimerText>
-              {getLocale('braveWalletConnectHardwareAuthorizationNeeded')}
-            </DisclaimerText>
-          </HardwareInfoColumn>
-        </HardwareInfoRow>
-        <AuthorizeHardwareDeviceIFrame coinType={selectedAccountType.coin} />
+        <HardwareWalletAccountsList
+          hardwareWallet={selectedHardwareWallet}
+          accounts={accounts}
+          preAddedHardwareWalletAccounts={preAddedHardwareWalletAccounts}
+          onLoadMore={onSubmit}
+          selectedDerivationPaths={selectedDerivationPaths}
+          setSelectedDerivationPaths={setSelectedDerivationPaths}
+          selectedDerivationScheme={selectedDerivationScheme}
+          setSelectedDerivationScheme={onChangeDerivationScheme}
+          onAddAccounts={onAddAccounts}
+          filecoinNetwork={filecoinNetwork}
+          onChangeFilecoinNetwork={onFilecoinNetworkChanged}
+          coin={selectedAccountType.coin}
+        />
       </>
     )
   }
 
-  if (showAccountsList) {
+  if (!selectedHardwareWallet) {
     return (
-      <HardwareWalletAccountsList
-        hardwareWallet={selectedHardwareWallet}
-        accounts={accounts}
-        preAddedHardwareWalletAccounts={preAddedHardwareWalletAccounts}
-        onLoadMore={onSubmit}
-        selectedDerivationPaths={selectedDerivationPaths}
-        setSelectedDerivationPaths={setSelectedDerivationPaths}
-        selectedDerivationScheme={selectedDerivationScheme}
-        setSelectedDerivationScheme={onChangeDerivationScheme}
-        onAddAccounts={onAddAccounts}
-        filecoinNetwork={filecoinNetwork}
-        onChangeFilecoinNetwork={onFilecoinNetworkChanged}
-        coin={selectedAccountType.coin}
-      />
+      <>
+        <HardwareButton
+          title={getLocale('braveWalletConnectHardwareLedger')}
+          description={getLocale(
+            'braveWalletConnectHardwareDeviceDescription'
+          ).replace('$1', getLocale('braveWalletConnectHardwareLedger'))}
+          onClick={onSelectLedger}
+        />
+        <Divider />
+        <HardwareButton
+          title={getLocale('braveWalletConnectHardwareTrezor')}
+          description={getLocale(
+            'braveWalletConnectHardwareDeviceDescription'
+          ).replace('$1', getLocale('braveWalletConnectHardwareTrezor'))}
+          onClick={onSelectTrezor}
+        />{' '}
+      </>
```

Comment:
move this above `if (showAccountsList && selectedHardwareWallet) {` so that we can remove `&& selectedHardwareWallet) from that line